components/compliance-service/reporting/relaxting/reports.go

```diff
@@ -530,6 +530,212 @@ func (backend *ES2Backend) GetReport(esIndex string, reportId string,
+		return nil, err
+	}
+
+	// we need to use start_time or else we won't see all of the controls that were used outside of end_time day
+	// we use start_time for the same reason we need it for trends
+	esIndex, err := GetEsIndex(filters, false, true)
+	if err != nil {
+		return nil, errors.Wrap(err, myName)
+	}
+
+	//here, we set latestOnly to true.  We may need to set it to false if we want to search non lastest reports
+	//for now, we don't search non-latest reports so don't do it.. it's slower for obvious reasons.
+	filtQuery := backend.getFiltersQuery(filters, true)
+
+	searchSource := elastic.NewSearchSource().
+		Query(filtQuery).
+		FetchSource(false).
+		Size(1)
+
+	controlTermsAgg := elastic.NewTermsAggregation().Field("profiles.controls.id").
+		Size(100).
+		Order("_key", true)
+
+	controlTermsAgg.SubAggregation("impact",
+		elastic.NewTermsAggregation().Field("profiles.controls.impact").Size(1))
+
+	controlTermsAgg.SubAggregation("title",
+		elastic.NewTermsAggregation().Field("profiles.controls.title").Size(1))
+
+	controlTermsAgg.SubAggregation("skipped",
+		elastic.NewFilterAggregation().Filter(elastic.NewTermQuery("profiles.controls.status", "skipped")))
+
+	controlTermsAgg.SubAggregation("failed",
+		elastic.NewFilterAggregation().Filter(elastic.NewTermQuery("profiles.controls.status", "failed")))
+
+	controlTermsAgg.SubAggregation("passed",
+		elastic.NewFilterAggregation().Filter(elastic.NewTermQuery("profiles.controls.status", "passed")))
+
+	controlTermsAgg.SubAggregation("end_time", elastic.NewReverseNestedAggregation().SubAggregation("most_recent_report",
+		elastic.NewTermsAggregation().Field("end_time").Size(1)))
+
+	profileInfoAgg := elastic.NewReverseNestedAggregation().Path("profiles").SubAggregation("version",
+		elastic.NewTermsAggregation().Field("profiles.name"))
+
+	profileInfoAgg.SubAggregation("sha",
+		elastic.NewTermsAggregation().Field("profiles.sha256"))
+
+	profileInfoAgg.SubAggregation("title",
+		elastic.NewTermsAggregation().Field("profiles.title"))
+
+	profileInfoAgg.SubAggregation("version",
+		elastic.NewTermsAggregation().Field("profiles.version"))
+
+	controlTermsAgg.SubAggregation("profile", profileInfoAgg)
+
+	//todo handle passed in regex too
+	filteredControls := elastic.NewFilterAggregation().Filter(elastic.NewBoolQuery().Must(elastic.NewRegexpQuery("profiles.controls.id", ".*")))
+
+	filteredControls.SubAggregation("control", controlTermsAgg)
+
+	controlsAgg := elastic.NewNestedAggregation().Path("profiles.controls")
+	controlsAgg.SubAggregation("filtered_controls", filteredControls)
+
+	searchSource.Aggregation("controls", controlsAgg)
+
+	source, err := searchSource.Source()
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("%s unable to get Source", myName))
+	}
+	LogQueryPartMin(esIndex, source, fmt.Sprintf("%s query", myName))
+
+	searchResult, err := client.Search().
+		SearchSource(searchSource).
+		Index(esIndex).
+		Do(context.Background())
```

Comment:
Should we thread the context from the request through to here?

Reply:
Good point.  I see a lot of places where we need to thread the context through.  We should do a pr just for that in all necessary places.

Reply:
done